make/compiler_flags

```diff
@@ -306,9 +311,20 @@ ifdef STAN_MPI
   CXXFLAGS_MPI ?= -Wno-delete-non-virtual-dtor
 endif
 
+################################################################################
+# Setup Sundials
+#
+# Add compiler flags to link to installed versions of Sundials
+
+ifdef SYS_LIBS
```

Comment:
A minor thing: would you mind having the `ifdef` go in the same direction? It'll just make it easier when coming across this code. (some of these are `ifdnef SYS_LIBS` and some are `ifdef SYS_LIBS`)

I'd prefer it to be `ifdef SYS_LIBS`, but that's just a preference.